src/c++/perf_analyzer/client_backend/client_backend.h

```diff
@@ -35,53 +35,20 @@
 #include <string>
 #include <vector>
 #include "ipc.h"
-
-#define RETURN_IF_CB_ERROR(S)    \
-  do {                           \
-    const Error& status__ = (S); \
-    if (!status__.IsOk()) {      \
-      return status__;           \
-    }                            \
-  } while (false)
-
+#include "../error.h"
 namespace perfanalyzer { namespace clientbackend {
-
-//==============================================================================
-/// Error status reported by backends
-///
-class Error {
- public:
-  /// Create an error with the specified message.
-  /// \param msg The message for the error
-  explicit Error(const std::string& msg = "");
-
-  /// Accessor for the message of this error.
-  /// \return The messsage for the error. Empty if no error.
-  const std::string& Message() const { return msg_; }
-
-  /// Does this error indicate OK status?
-  /// \return True if this error indicates "ok"/"success", false if
-  /// error indicates a failure.
-  bool IsOk() const { return msg_.empty(); }
-
-  /// Convenience "success" value. Can be used as Error::Success to
-  /// indicate no error.
-  static const Error Success;
-
- private:
-  friend std::ostream& operator<<(std::ostream&, const Error&);
-  std::string msg_;
-};
-
-//===================================================================================
-
 class ClientBackend;
 class InferInput;
 class InferRequestedOutput;
 class InferResult;
-
-enum BackendKind { TRITON = 0, TENSORFLOW_SERVING = 1, TORCHSERVE = 2 };
-enum ProtocolType { HTTP = 0, GRPC = 1, UNKNOWN = 2 };
+class TritonLoader;
+enum BackendKind {
+  TRITON = 0,
+  TENSORFLOW_SERVING = 1,
+  TORCHSERVE = 2,
+  TRITON_LOCAL = 3
+};
+enum ProtocolType { HTTP = 0, GRPC = 1, UNKNOWN = 2, LOCAL = 3 };
```

Comment:
Why is LOCAL needed here? I assume triton_c_api will just ignore the protocol setting.

Reply:
yep will remove this